index.js

```diff
@@ -13,22 +13,25 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  const name = options.name
+  delete options.name
+
   const pool = new pg.Pool(options)
 
-  fastify.decorate('pg', {
+  if (!fastify.pg) {
+    fastify.decorate('pg', {})
+  }
+
+  fastify.pg[name] = {
     connect: pool.connect.bind(pool),
     pool: pool,
     Client: pg.Client,
     query: pool.query.bind(pool)
-  })
+  }
 
-  fastify.addHook('onClose', onClose)
+  fastify.addHook('onClose', (fastify, done) => pool.end(done))
```

Comment:
Why this change?
The reason of the separated function is that if something goes wrong its easier understand which function has failed, since it has a name ;)

Reply:
1. I had tried `name` configurable first time, but I think maybe it makes some user confuesed. sometimes `pg` is an connection container and sometimes it is an connection. I can not make a choice myself. If you think `name` configuable better, I could do it.

2. using closure because I need refer the connection pool. If I still use `onClose` function, I cloud not get the pool to end.